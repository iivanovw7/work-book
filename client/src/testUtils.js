import 'cross-fetch/polyfill';
import 'jsdom-global/register';
import { mount, shallow } from 'enzyme';

import { createMemoryHistory } from 'history';
import * as localized from './assets/locales/locales.json';

// Mock browser history
export const history = createMemoryHistory('/dashboard');

// Grid config object for testing
export const mappedSizes = {
  xs: 12,
  sm: 12,
  md: 12,
  lg: 11
};

export const localizedText = {
  locales: localized.locales,
  topBar: localized.topBar,
  login: localized.login,
  navigation: localized.navigation,
  pageNotFound: localized.pageNotFound,
  posts: localized.posts,
  search: localized.search
};

// Suppresses console warnings until component is mounted
export function suppressConsoleWarnings(component, mountType, consoleObject) {
  const originalError = consoleObject.error;
  const originalWarn = consoleObject.warn;
  // eslint-disable-next-line no-param-reassign
  consoleObject.error = jest.fn();
  // eslint-disable-next-line no-param-reassign
  consoleObject.warn = jest.fn();

  const createdComponent = mountType === 'mount' ? mount(component) : shallow(component);
  // eslint-disable-next-line no-param-reassign
  consoleObject.error = originalError;
  // eslint-disable-next-line no-param-reassign
  consoleObject.warn = originalWarn;

  return createdComponent;
}

export const state = {
  locale: 'eng',
  theme: 'dark',
  search: ''
};
