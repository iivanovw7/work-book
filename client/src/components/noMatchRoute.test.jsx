import React from 'react';
import { mount } from 'enzyme';
import chalk from 'chalk';
import useStoreon from 'storeon/react';
import { MemoryRouter } from 'react-router-dom';
import NoMatchRoute from './noMatchRoute';
import * as testUtils from '../testUtils';

jest.mock('../config/apiURL', () => () => '/api');
jest.mock('../config/settings', () => ({
  ...(jest.requireActual('../config/settings'))
}));
jest.mock('storeon/react', () => jest.fn());

// Used in order to return "localizedText" variable which is out of scope
useStoreon.mockImplementation(() => ({
  locale: 'eng',
  theme: 'dark',
  localizedText: testUtils.localizedText,
  tags: 'tags,tags'
}));

describe(`Testing [${chalk.yellow('NoMatchRoute')}] component`, () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  const props = {
    history: testUtils.history,
    message: 'message',
    location: {
      pathname: 'pathname'
    }
  };

  const Composition = () => (
    <MemoryRouter>
      <NoMatchRoute userAccess {...props} />
    </MemoryRouter>
  );

  it('Should match snapshot', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition handleClick={mockFunc()} />);
    expect(component).toMatchSnapshot();
  });

  it('Should render with correct props', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition handleClick={mockFunc()} />);
    expect(component.find('ins')).toHaveLength(1);
    expect(component.find('ins').text()).toBe(props.location.pathname);
  });

  it('Should trigger link click event', () => {
    const mockFunc = jest.fn();
    const component = mount(<Composition handleClick={mockFunc()} />);
    expect(component.find('a')).toHaveLength(1);
    component.find('a').at(0).simulate('click');
    expect(mockFunc).toHaveBeenCalled();
  });
});
