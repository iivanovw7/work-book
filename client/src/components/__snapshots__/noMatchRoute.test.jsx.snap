// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing [[33mNoMatchRoute[39m] component Should match snapshot 1`] = `ReactWrapper {}`;
